### ISSUE TITLE
Problem with psalter using UnusedVariable issue and variable assigned by reference. 

### ISSUE THREAD

**Issue by ADmad** [vimeo/psalm#2423]
Running `psalm --alter --issues=UnusedVariable` it incorrectly removes the following line:

https://github.com/cakephp/cakephp/blob/7751403267ba4c8196cc444d70ac3877eac0911b/src/Core/InstanceConfigTrait.php#L275

No error is reported for that line when just using `pslam`.

**Comment by muglug** [comment]
Oh, by-reference variables. Fun!

**Comment by ADmad** [comment]
You should start using the CakePHP codebase as a test suite for psalm :smile:.

**Comment by muglug** [comment]
Reproduced here: https://psalm.dev/r/facfc66813

```php
<?php
class A {
  public string $value = '';
  public function writeByRef(string $value): void {
    $update =& $this->value;
    $update = $value;
  }
}
```
